lib/cocoapods/installer/analyzer.rb

```diff
@@ -306,6 +306,7 @@ def generate_pod_targets(specs_by_target)
           # transitive dependencies can't be deduplicated.
           pod_targets.flat_map do |target|
             dependent_targets = transitive_dependencies_for_pod_target(target, pod_targets)
+            target.dependent_targets = dependent_targets
```

Comment:
this is recursive transitive dependencies, rather than shallow dependencies. Is this OK?